internal/auth/oidc/repository_account_test.go

```diff
@@ -0,0 +1,800 @@
+package oidc
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/boundary/internal/auth/oidc/store"
+	"github.com/hashicorp/boundary/internal/db"
+	dbassert "github.com/hashicorp/boundary/internal/db/assert"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/iam"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRepository_LookupAccount(t *testing.T) {
+	conn, _ := db.TestSetup(t, "postgres")
+	rw := db.New(conn)
+	wrapper := db.TestWrapper(t)
+
+	kmsCache := kms.TestKms(t, conn, wrapper)
+	iamRepo := iam.TestRepo(t, conn, wrapper)
+	org, _ := iam.TestScopes(t, iamRepo)
+
+	ctx := context.Background()
+	databaseWrapper, err := kmsCache.GetWrapper(ctx, org.PublicId, kms.KeyPurposeDatabase)
+	require.NoError(t, err)
+
+	authMethod := TestAuthMethod(
+		t, conn, databaseWrapper, org.PublicId, ActivePrivateState,
+		TestConvertToUrls(t, "https://www.alice.com")[0],
+		"alice-rp", "fido",
+		WithSigningAlgs(RS256),
+		WithCallbackUrls(TestConvertToUrls(t, "https://www.alice.com/callback")[0]),
+	)
+	issuerId := TestConvertToUrls(t, authMethod.DiscoveryUrl)[0]
+	account := TestAccount(t, conn, authMethod, issuerId, "test-subject")
+
+	newAcctId, err := newAccountId(authMethod, issuerId.String(), "random-id")
+	require.NoError(t, err)
+	tests := []struct {
+		name       string
+		in         string
+		want       *Account
+		wantIsErr  errors.Code
+		wantErrMsg string
+	}{
+		{
+			name:       "With no public id",
+			wantIsErr:  errors.InvalidPublicId,
+			wantErrMsg: "oidc.(Repository).LookupAccount: missing public id: parameter violation: error #102",
+		},
+		{
+			name: "With non existing account id",
+			in:   newAcctId,
+		},
+		{
+			name: "With existing account id",
+			in:   account.GetPublicId(),
+			want: account,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			assert, require := assert.New(t), require.New(t)
+			repo, err := NewRepository(rw, rw, kmsCache)
+			assert.NoError(err)
+			require.NotNil(repo)
+			got, err := repo.LookupAccount(context.Background(), tt.in)
+			if tt.wantIsErr != 0 {
+				assert.Truef(errors.Match(errors.T(tt.wantIsErr), err), "Unexpected error %s", err)
+				assert.Equal(tt.wantErrMsg, err.Error())
+				return
+			}
+			require.NoError(err)
+			assert.EqualValues(tt.want, got)
+		})
+	}
+}
+
+func TestRepository_DeleteAccount(t *testing.T) {
```

Comment:
consider adding a test for missing scope id.

Reply:
Added. Thanks!